classes/booking_manager.php

```diff
@@ -403,9 +454,15 @@ public function process() {
                 // Map status to status code.
                 $statuscode = array_search($entry->status, facetoface_statuses()) ?: MDL_F2F_STATUS_BOOKED;
 
+                $signupstatuses = [MDL_F2F_STATUS_BOOKED, MDL_F2F_STATUS_WAITLISTED];
+                $attendancestatuses = [MDL_F2F_STATUS_NO_SHOW, MDL_F2F_STATUS_PARTIALLY_ATTENDED, MDL_F2F_STATUS_FULLY_ATTENDED];
+
+                $issignup = in_array($statuscode, $signupstatuses);
+                $isattendance = in_array($statuscode, $attendancestatuses);
+
                 // Handle signups.
-                if (in_array($statuscode, [MDL_F2F_STATUS_BOOKED, MDL_F2F_STATUS_WAITLISTED])) {
-                    if ($statuscode === MDL_F2F_STATUS_BOOKED && !$session->datetimeknown) {
+                if ($issignup) {
+                    if (($statuscode === MDL_F2F_STATUS_BOOKED) && !$session->datetimeknown) {
```

Comment:
```suggestion
                    if ($statuscode === MDL_F2F_STATUS_BOOKED && !$session->datetimeknown) {
```
Unnecessary grouping